sdk/src/androidTest/java/com/mapbox/maps/StyleLoadTest.kt

```diff
@@ -35,94 +36,114 @@ class StyleLoadTest {
       }
     }
-    countDownLatch.await(10, TimeUnit.SECONDS)
-    Assert.assertEquals(2, styleList.size)
-    assertTrue(styleList[0].isValid())
-    assertTrue(styleList[1].isValid())
+    countDownLatch.throwExceptionOnTimeoutMs()
+
+    assertFalse(styles[0].isValid())
+    assertTrue(styles[1].isValid())
+
     mapboxMap.onDestroy()
-    assertFalse(styleList[0].isValid())
-    assertFalse(styleList[1].isValid())
+
+    assertFalse(styles[1].isValid())
   }
 
   @Test
   fun testStyleAsyncGetter() {
-    countDownLatch = CountDownLatch(1)
-    var styleLoadedCount = 0
+    countDownLatch = CountDownLatch(2)
     rule.scenario.onActivity {
       it.runOnUiThread {
         mapboxMap.getStyle { style ->
-          assertNotNull("Style should but non null", style)
           assertTrue("Style should be fully loaded", style.isStyleLoaded)
-          styleLoadedCount++
+          countDownLatch.countDown()
         }
 
         mapboxMap.loadStyleUri(
           Style.MAPBOX_STREETS
         ) { style ->
-          assertNotNull("Style should but non null", style)
           assertTrue("Style should be fully loaded", style.isStyleLoaded)
-          styleLoadedCount++
           countDownLatch.countDown()
         }
         mapView.onStart()
       }
     }
-    countDownLatch.await(10, TimeUnit.SECONDS)
-    Assert.assertEquals(2, styleLoadedCount)
+    countDownLatch.throwExceptionOnTimeoutMs(10_000)
   }
 
   @Test
-  fun testFullyLoadedFalse() {
+  fun testStyleIsLoadedOnStyleChange() {
     rule.scenario.onActivity {
       it.runOnUiThread {
         mapboxMap.loadStyleUri(
           Style.MAPBOX_STREETS
         ) { style ->
-          assertNotNull("Style should but non null", style)
           assertTrue("Style should be fully loaded", style.isStyleLoaded)
-          mapboxMap.loadStyleUri(Style.SATELLITE)
+          mapboxMap.loadStyleUri(Style.SATELLITE) {
+            assertTrue("Style should be fully loaded again", style.isStyleLoaded)
+            countDownLatch.countDown()
+          }
           assertFalse("Map shouldn't be fully loaded", style.isStyleLoaded)
-          countDownLatch.countDown()
         }
         mapView.onStart()
       }
     }
+    countDownLatch.throwExceptionOnTimeoutMs(30_000)
+  }
+
+  @Test
+  fun testLoadMultipleStylesInARow() {
     countDownLatch = CountDownLatch(1)
-    if (!countDownLatch.await(30, TimeUnit.SECONDS)) {
-      throw TimeoutException()
+    rule.scenario.onActivity {
+      it.runOnUiThread {
+        mapboxMap.loadStyleUri(Style.MAPBOX_STREETS)
+        mapboxMap.loadStyle(style("") {})
+        mapboxMap.loadStyleUri(Style.SATELLITE)
+        mapboxMap.loadStyleUri(Style.MAPBOX_STREETS)
+        mapboxMap.loadStyle(style("") {})
+        mapboxMap.loadStyleUri(Style.MAPBOX_STREETS)
+        mapboxMap.loadStyleUri(Style.SATELLITE)
+        mapboxMap.loadStyleUri(Style.MAPBOX_STREETS) { style ->
+          assertTrue("Style should be fully loaded", style.isStyleLoaded)
```

Comment:
also could check that all previous styles are not valid and the only valid one is the last one

Reply:
We won't receive any styles in the previous cases.
Added counter to check that.